tests/testthat/test-session.property.R

```diff
@@ -0,0 +1,54 @@
+# Copyright (c) 2015-present, Facebook, Inc.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+
+context("session.property")
+
+source('utilities.R')
+
+query_session_property <- function(conn, property) {
+  properties <- dbGetQuery(conn, 'SHOW SESSION')
+  properties[properties$Name == property, ]$Value
+}
+
+test_that('session properties can be configured in connection', {
+  conn <- setup_live_connection()
+  expect_equal(conn@parameters, list())
+
+  # verify default value is applied
+  test_property <- 'optimize_hash_generation'
+  expect_equal(query_session_property(conn, test_property), 'true')
```

Comment:
Even though it's a rather safe assumption, this default value would not be guaranteed. A better option would be to take the default value and invert it in the following tests.